### PR TITLE
clean up to support partial deserialization

### PR DESCRIPTION
**Pull request by jeanphilippeD**

### CODE REVIEW

File: src/messages/mod.rs [hunk 1]
```diff
@@ -68,6 +67,8 @@ pub struct HopMessageWithBytes {
     content: SignedRoutingMessage,
     /// Serialized Message as received or sent to quic_p2p.
     full_message_bytes: Bytes,
+    /// Crypto hahs of the full message.
+    full_message_crypto_hash: Digest,
```
Comment (madadam) [inline src/messages/mod.rs]: Minor, but do we need to include `crypto` in the name? Can we just assume that "hash" means "crypto hash:?

File: src/messages/mod.rs [hunk 2]
```diff
@@ -68,6 +67,8 @@ pub struct HopMessageWithBytes {
     content: SignedRoutingMessage,
     /// Serialized Message as received or sent to quic_p2p.
     full_message_bytes: Bytes,
+    /// Crypto hahs of the full message.
```
Comment (madadam) [inline src/messages/mod.rs]: Typo: "Crypto hash..."

File: src/messages/mod.rs [hunk 3]
```diff
@@ -31,6 +31,8 @@ use std::{
     mem,
 };
 
+type Digest = [u8; 32];
```
Comment (madadam) [inline src/messages/mod.rs]: This is already defined in the `crypto` module. It's called `Digest256` though.

File: src/messages/mod.rs (hunk 1 above)
Comment (jeanphilippeD) [reply src/messages/mod.rs]: not really, we have Hash derive a lot and the .hash function is not crypto. I'd feel a bit more comfortable keeping it.

**Comment by jeanphilippeD** [comment]
Fixed review comments

**Pull request merged**